### DISCUSSION TITLE
Does Nats (c#) support open telemetry?

### DISCUSSION THREAD

**Discussion by LightVolk** [nats-io/nats.net.v1#720] [Q&A] [answered]
I try to make trace with nats+jaeger , found some reps called not.go and not.java , nut cannot find c# realization.

For example - for java - [https://github.com/nats-io/not.java](url)

**Comment by scottf** [comment]
@LightVolk I'm not sure what you are asking and the url you posted is not working.

**Comment by ColinSullivan1** [comment, marked as answer]
Unfortunately we do not have an Open Telemetry reference architecture for .NET today.

However, I'd suggest propagating spans / traces across headers as described here:  https://stackoverflow.com/a/74477063. 

At the time not.java and not.go was developed we were still developing headers, so we used the binary formatters. Since NATS supports headers today, I'd definitely suggest using headers as in the example above.

An open source C# wrapper/reference architecture for open tracing would be a contribution we would welcome!
